internal/policies/fetcher.go

```diff
@@ -109,6 +105,24 @@ func (f *Fetcher) GetNamespace(nsName string) (*v1.Namespace, error) {
 	return namespace, nil
 }
 
+// GetAuditedNamespaces gets all namespaces besides the ones in fetcher.skippedNs
+// This function cannot be tested with fake.client, as fake.client doesn't
+// support fields.OneTermNotEqualSelector()
+func (f *Fetcher) GetAuditedNamespaces() (*v1.NamespaceList, error) {
+	skipNsFields := fields.Everything()
+	for _, nsName := range f.skippedNs {
+		skipNsFields = fields.AndSelectors(skipNsFields, fields.OneTermNotEqualSelector("metadata.name", nsName))
+		log.Debug().Str("ns", nsName).Msg("skipping ns")
+	}
```

Comment:
Can't you do the equivalent of:

```yaml
  namespaceSelector:
     matchExpressions:
     - key: "kubernetes.io/metadata.name"
       operator: NotIn
       values: [kube-system, cattle-monitoring-system]
```

From my understanding you are building a long `(name != FOO) && (name != BAR) && ...)`

Reply:
I found no way to do it differently. All functions only accept a string (so, 1 field value), and compare 1 key to 1 field.

One could do `fields.ParseSelector("metadata.name!=" + "default")` and build a SCV of key and fields to select for. I have seen code examples for that in the wild. But looking at how they get parsed in the underlying field code yields the same implementation, plus one then needs to sanitize the namespace names so there can't be an attack when building that string to parse as selector.